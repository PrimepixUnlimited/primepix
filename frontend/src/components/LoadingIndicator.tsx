import React, { FC } from 'react'
import { ActivityIndicator, Text, View, StyleSheet } from 'react-native'

interface Props {
  loading: boolean
}

import styles from '../constants/styles'

const LoadingIndicator: FC<Props> = ({ loading = false }) =>
  loading && (
    <View style={s.container}>
      <ActivityIndicator size="small" color={styles.colors.primary} />
      <Text style={styles.text.body}>Loading...</Text>
    </View>
  )

export default LoadingIndicator

const s = StyleSheet.create({
  container: {
    flexDirection: 'row'
  }
})
