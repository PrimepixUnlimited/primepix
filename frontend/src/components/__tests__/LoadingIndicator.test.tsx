// components/__tests__/Hello.tsx
import "react-native";
import * as React from "react";
import * as renderer from "react-test-renderer";

import LoadingIndicator from "../LoadingIndicator";

describe("LoadingIndicator snapshot", () => {
  it("renders correctly", () => {
    const component = renderer.create(<LoadingIndicator loading />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it("hides correctly", () => {
    const component = renderer
      .create(<LoadingIndicator loading={false} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
