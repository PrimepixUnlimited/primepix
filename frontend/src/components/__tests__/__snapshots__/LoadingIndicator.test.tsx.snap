// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator snapshot hides correctly 1`] = `null`;

exports[`LoadingIndicator snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#2cd4c8"
    hidesWhenStopped={true}
    size="small"
  />
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
        "lineHeight": 30,
      }
    }
  >
    Loading...
  </Text>
</View>
`;
